app/_hub/kong-inc/graphql-rate-limiting-advanced/1.3.x.md

```diff
@@ -0,0 +1,575 @@
+
+$ curl -X POST http://kong:8001/services/{service}/graphql-rate-limiting-advanced/costs \
+  --data type_path="Vehicle.filmConnection" \
+  --data mul_arguments="first"
+
+
+$ curl -X POST http://kong:8001/services/{service}/graphql-rate-limiting-advanced/costs \
+  --data type_path="Film.characterConnection" \
+  --data mul_arguments="first"
+```
+
+
+#### `/graphql-rate-limiting-advanced/costs`
+
+* **GET**: list of all costs on any service
+* **PUT**, **POST**: add a cost to a service schema
+
+
+#### `/graphql-rate-limiting-advanced/costs/{cost_id}`
+
+* **GET**: get cost associated by id
+* **PATCH**: modify cost associated by id
+* **DELETE**: delete cost associated by id
+
+
+### Changing the default strategy
+
+```
+$ curl -X POST http://kong:8001/services/{service}/plugins \
+  --data name=graphql-rate-limiting-advanced \
+  --data config.limit=100 \
+  --data config.limit=10000 \
+  --data config.window_size=60 \
+  --data config.window_size=3600 \
+  --data config.sync_rate=10 \
+  --data config.cost_strategy=default
+```
+
+```
+$ curl -i -X PATCH http://kong:8001/plugins/{plugin_id} \
+  --data config.cost_strategy=node_quantifier
+```
+
+### Limit query cost by using `config.max_cost`
+
+It's usually a good idea to define a maximum cost applied to any query,
+regardless if the call is within the rate limits for a consumer.
+
+By defining a `max_cost` on our service, we are ensuring no query will run with
+a cost higher than our set `max_cost`. By default it's set to 0, which means
+no limit.
+
+```
+$ curl -X POST http://kong:8001/services/{service}/plugins \
+  --data name=graphql-rate-limiting-advanced \
+  --data config.limit=100 \
+  --data config.limit=10000 \
+  --data config.window_size=60 \
+  --data config.window_size=3600 \
+  --data config.sync_rate=0 \
+  --data config.max_cost=5000
+```
+
+If `max_cost` needs to be updated, one can accomplish this by doing the following:
+
+```
+$ curl -X PATCH http://kong:8001/plugins/{plugin_id} \
+  --data config.max_cost=10000
+```
+
+### Using `config.score_factor` to modify costs
+
+GraphQL query cost depends on multiple factors based on our resolvers
+and the implementation of the schema. Cost on queries, depending on the cost
+strategy might turn very high when using quantifiers, or very low with no
+quantifiers at all. By using `config.score_factor` the cost can be divided
+or multiplied to a certain order of magnitude.
+
+For example, a `score_factor` of `0.01` will divide the costs by 100, meaning
+every cost unit represents 100 nodes.
+
+```
+$ curl -X POST http://kong:8001/services/{service}/plugins \
+  --data name=graphql-rate-limiting-advanced \
+  --data config.limit=100 \
+  --data config.limit=10000 \
+  --data config.window_size=60 \
+  --data config.window_size=3600 \
+  --data config.sync_rate=0 \
+  --data config.max_cost=5000 \
+  --data config.score_factor=0.01
+```
+
+If `score_factor` needs to be updated, one can accomplish this by doing the following:
```

Comment:
```suggestion
To update `score_factor`:
```

Reply:
@HCloward yeah that's pretty wordy. I'm trying not to scope creep beyond adding datatypes/missing fields but this begs to be edited. So this file is not new, it is the old one that used to be the index. When a new index file gets created, it shows the prior one as a new file. It also bloats the repo.

I'll make these edits to the old and new index file. Please just look at the index file in future PRs. I'll be sure to add that to the reviewer descriptions not to bother with old source.